tests/Integration/config/5.4/routes/controllers.yaml

```diff
@@ -0,0 +1,26 @@
+archive:
```

Comment:
what are these `tests/Integration/config/*/routes/controllers.yaml` files for ?

Reply:
As I dropped the usage of `sensio/framework-extra-bundle` which is also 'end-of-life' since Syfmony 6.2 ([see](https://github.com/sensiolabs/SensioFrameworkExtraBundle)). 

I had to refactor the use of routing annotations (which was part of that bundle), and as routing attributes are not supported on lower versions I refactored the routing annotations to `yaml` routing config. So these `controllers.yaml` files contain the `yaml` routing definitions.

Reply:
So it means that we are not able to test that on symfony 5.4 the bundle is still able to work with annotation  
This might be an issue IMO

Reply:
We aren't explicitly testing `Route`-annotations anymore indeed, but the only change made is the way the routing is defined. Both of the syntaxes result in the same underlying `Route`-object.

For example 

```php
/**
 * @Route("/blog", name="blog_read", options={"sitemap"={"section"="blog"}})
*/
```

Is simply rewritten to the `Yaml` syntax
``` yaml
blog_read:
    path: /blog
    controller: Presta\SitemapBundle\Tests\Integration\Controller\BlogController::read
    options:
        sitemap:
            section: blog
```

Since the bundle ([sensio/framework-extra-bundle](https://packagist.org/packages/sensio/framework-extra-bundle)) is abandoned, it's very unlikely that the currently used syntax wouldn't work anymore (for projects using Symfony `<6.3`). And even if it would turn out to be erroneous it would be related to the same code in `symfony/routing` as the `Yaml` or `Attribute` syntax uses and thus be still supported in our testsuite.

If we would have had an own `Annotation`-route class depending on the bundle above that would be something different, the defined 'options' we use are just part of the same code as before.

So we are not compromising on our testsuite, we just use an alternative syntax to define the same arrangement to build the testsuite.

Reply:
Hi Benoit,

Thanks for reaching out, first of all I never used annotations to configure my routes myself, so I had to dive into it all too, and this is what I got out of it.

You both are right, I went off-track with the notification on the [SensioFrameworkExtraBundle](https://symfony.com/bundles/SensioFrameworkExtraBundle/current/index.html) docs, and mistakenly assumed Annotations were completely dropped in higher versions, which would make still supporting it on all versions problematic.

> It's no longer recommended to use this bundle in current Symfony applications. All the annotations provided by this bundle are now built-in in Symfony as PHP attributes. Check out the [full list of Symfony attributes](https://symfony.com/doc/current/reference/attributes.html).

They don't mention Annotations are still supported, but [here](https://symfony.com/bundles/SensioFrameworkExtraBundle/current/annotations/routing.html) they do :)

I've reverted the changes and it looks like the tests still pass.

Things to note:
- if I'm correct we are now only testing the Annotations routes on Symfony `5.4` and PHP `< 8.0`, hence the check in [Kernel.php](https://github.com/prestaconcept/PrestaSitemapBundle/blob/3.x/tests/Integration/src/Kernel.php#L51-L53)

Let's see if the workflow still passes 🤞

Reply:
No worries, it's quite confusing anyway that Symfony started to use the same `@Route` annotation's name in replacement of Sensio's one.

Here is the trick:
```
-use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Symfony\Component\Routing\Annotation\Route;
```
The first import comes from Sensio's bundle and is deprecated, the second one comes from Symfony and is perfectly valid to be used as annotation `@Route` or attribute `#[Route]`.

Thank you for having taken this into account 🙂